docs/getting-started/cli.md

```diff
@@ -0,0 +1,198 @@
+
+To view the results in the CLI:
+
+```console
+semgrep ci
+```
+
+To export the results to a plain text file:
+
+```console
+semgrep ci --text --output=semgrep.txt
+```
+
+To export the results to a SARIF file:
+
+```console
+semgrep ci --sarif --output=semgrep.sarif
+```
+
+To export the results to a JSON file:
+
+```console
+semgrep ci --json --output=semgrep.json
+```
+
+### Scan your project with a specific ruleset
+
+You can scan your project using `--config auto` to automatically obtain rules tailored to your project:
+
+```console
+semgrep scan --config auto
+```
+
+:::info
+Semgrep collects pseudonymous metrics when you use rules from the Registry.
+:::
+
+To scan your project with a specific ruleset, either one that you write or one that you obtain from the [Semgrep Registry](https://semgrep.dev/explore), you can do so using the `--config` flag:
+
+```console
+# Scan with the rules defined in rules.yaml
+semgrep scan --config rules.yaml
+```
+
+You can include as many configuration flags as necessary.
+
+```console
+# Scan with rules defined in two separate config files
+semgrep scan --config rules.yaml --config more_rules.yaml
+```
+
+#### Test custom rules
+
+Semgrep includes functionality to [test the custom rules that you write](/writing-rules/testing-rules/):
+
+```console
+semgrep scan --test
+```
+
+#### Publish custom rules
+
+To share your rules by adding them to the Semgrep Registry:
+
+```console
+semgrep publish <path/to/rules>
+```
+
+### Scan without sending results to Semgrep
+
+To scan your project **without** sending data to Semgrep, use:
+
+```console
+semgrep ci --dry-run
+```
+
+This can be helpful to verify the results of a specific ruleset or to see how your findings change based on the rulesets you choose for your scans.
+
+### Scan using Semgrep's OSS engine
+
+To scan your project with Semgrep's OSS Engine, even though you have the Pro Engine enabled in Semgrep Cloud Platform:
+
+```console
+semgrep ci --oss-only
+```
+
+:::info
+See [Semgrep Pro versus Semgrep OSS](/semgrep-pro-vs-oss) for information on the differences between Semgrep Pro and OSS Engines.
+:::
+
+## Scan using specific Semgrep Products
+
+When you run `semgrep ci`, you scan your project with Semgrep Code, Semgrep Supply Chain, and Semgrep Secrets. To scan your project with just one product, run:
+
+```console
+# scan with Semgrep Code
+semgrep ci --code
+
+# scan with Semgrep Supply Chain
+semgrep ci --supply-chain
+
+# scan with Semgrep Secrets
+semgrep ci --secrets
+```
+
+## Extend timeout thresholds
+
+Depending on the file sizes in your project, you may need to increase the timeout threshold so that Semgrep doesn't time out before the scan completes. You can control this value using the `--timeout-threshold` flag, which refers to the maximum amount of time Semgrep spends scanning a single file. The default value is 30 seconds. Semgrep attempts to scan each file with this timeout value three times.
```

Comment:
```suggestion
Depending on the file sizes in your project, you may need to increase the timeout threshold so that Semgrep doesn't time out before the scan completes. You can control this value using the `--timeout` flag, which refers to the maximum amount of time Semgrep spends scanning a single file. The default value is 5 seconds. Semgrep attempts to scan each file with this timeout value three times.
```
The time length is controlled with --timeout (default is 5 seconds in recent versions), and the number of times (default 3) is controlled with --timeout-threshold, so you could also add information on that after the last sentence.